evennia/contrib/unixcommand.py

```diff
@@ -0,0 +1,211 @@
+        super(UnixCommand, self).__init__()
+
+        # Create the empty EvenniaParser, inheriting argparse.ArgumentParser
+        lines = dedent(self.__doc__.strip("\n")).splitlines()
+        description = lines[0].strip()
+        epilog = "\n".join(lines[1:]).strip()
+        self.parser = EvenniaParser(None, description, epilog, command=self)
+
+        # Fill the argument parser
+        self.init()
+
+    def init(self):
+        """
+        Configure the argument parser, adding in options.
+
+        Note:
+            This method is to be overridden in order to add options
+            to the argument parser.  Use `self.parser`, which contains
+            the `argparse.ArgumentParser`.  You can, for instance,
+            use its `add_argument` method.
+
+        """
+        pass
+
+    def func(self):
+        """Override to handle the command execution."""
+        pass
+
+    def get_help(self, caller, cmdset):
+        """
+        Return the help message for this command and this caller.
+
+        Args:
+            caller (Object or Player): the caller asking for help on the command.
+            cmdset (CmdSet): the command set (if you need additional commands).
+
+        Returns:
+            docstring (str): the help text to provide the caller for this command.
+
+        """
+        return self.parser.format_help()
+
+    def parse(self):
+        """
+        Process arguments provided in `self.args`.
+
+        Note:
+            You should not override this method.  Consider overriding
+            `init` instead.
+
+        """
+        try:
+            self.opts = self.parser.parse_args(shlex.split(self.args))
+        except ParseError as err:
+            msg = str(err)
+            if msg:
+                self.msg(msg)
+            raise InterruptCommand
+
+
+class ParseError(Exception):
+
+    """An error occurred during parsing."""
+
+    pass
+
+
+class EvenniaParser(argparse.ArgumentParser):
+
+    """A parser just for Evennia."""
+
+    def __init__(self, prog, description="", epilog="", command=None, **kwargs):
+        prog = prog or command.key
+        super(EvenniaParser, self).__init__(
+                prog=prog, description=description,
+                conflict_handler='resolve', add_help=False, **kwargs)
+        self.command = command
+        self.post_help = epilog
+        def n_exit(code=None, msg=None):
+            raise ParseError(msg)
+
+        self.exit = n_exit
+
+        # Replace the -h/--help
+        self.add_argument("-h", "--hel", nargs=0, action=HelpAction, help="display heeeelp")
+
+    def format_usage(self):
+        """Return the usage line."""
+        return raw(super(EvenniaParser, self).format_usage())
+
+    def format_help(self):
+        """Return the parser help, including its epilog."""
+        autohelp = raw(super(EvenniaParser, self).format_help())
+        return "\n" + autohelp + "\n" + self.post_help
+
+    def print_usage(self, file=None):
```

Comment:
These methods does not appear to be used. If they are included for convenience, they should have full doc strings. What does `file` do, how can it be used?